flask_project/campaign_manager/views.py

```diff
@@ -690,26 +845,128 @@ def find_attribution(map_url):
     return attribution
 
 
+def delete_campaign_data(campaign):
+    task_boundaries = campaign.taskBoundaries
+    for taskboundary in task_boundaries:
+        session.delete(taskboundary.id_team)
+        session.delete(taskboundary)
+    feature_types = campaign.featureTypes
+    for feature_type in feature_types:
+        session.delete(feature_type)
+    functions = campaign.functions
+    for function in functions:
+        session.delete(function)
+    session.commit()
+
+
+def fillCampaignData(campaign, data):
```

Comment:
This function is too long (83 lines) and is very hard to digest (read and understand). And there are no tests.
I see 5 session.commit(), I would refactor this function into 5 smaller functions (one for each commit).

From the Kartoza guidelines: If a method or function is longer than a single screen, it is probably a candidate for refactoring into smaller methods / functions. Writing smaller methods makes your code easier to read and to test.

Reply:
Thanks for pointing it out. I have refactored the code with each campaign attribute having its own save and delete function.